src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -3260,6 +3257,11 @@ private static bool AttributeTypeNamesMatch(Type attributeType, Type reflectedAt
                             // and is not required for the manifest
                             if (eventAttribute.Channel != EventChannel.None)
                             {
+                                if (!manifest.WillBuild)
```

Comment:
Why do we need the manifest here? I thought it was fine to have a channel without generating a manifest.

Reply:
You are correct. I will remove the change. I had reincorporated your change and was also confused at the time as to why you added the check. Now I understand it only existed because the manifest could be null. Am I correct?